### PR TITLE
Feature bitbucket

### PR DESCRIPTION
**Pull request by utkarsh-lambdatest**
# Issue Link
https://github.com/LambdaTest/test-at-scale/issues/55

# Description

Add support for Bitbucket Git provider
Integrated changes regarding bitbucket integration in the build pipeline

Fixes # (issue)

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

- [x] Ran multiple builds on bitbucket repo with different webhook

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: pkg/gitmanager/setup.go [hunk 1]
```diff
@@ -47,7 +47,12 @@ func (gm *gitManager) Clone(ctx context.Context, payload *core.Payload, cloneTok
 		return err
 	}
 
-	if err = os.Rename(repoName+"-"+commitID, global.RepoDir); err != nil {
+	filename := repoName + "-" + commitID
+	if payload.GitProvider == core.Bitbucket {
+		filename = repoItems[len(repoItems)-2] + "-" + repoName + "-" + commitID[:12]
+	}
```
Comment (saurabh-prakash) [inline pkg/gitmanager/setup.go:53]: this is cryptic. Can you write in comments an example value that'll be contained here? Also, won't it be better if we download the archive in a definite filename?

File: pkg/urlmanager/urlmanager.go [hunk 2]
```diff
@@ -10,14 +10,18 @@ import (
 )
 
 // GetDownloadURL returns file download url for given git provider
-func GetDownloadURL(gitprovider, repoSlug, commitID, fileName string) (string, error) {
+func GetDownloadURL(gitprovider, repoLink, repoSlug, commitID, fileName string) (string, error) {
```
Comment (saurabh-prakash) [inline pkg/urlmanager/urlmanager.go:13]: remove this function. It is not being used.

Comment (utkarsh-lambdatest) [reply pkg/urlmanager/urlmanager.go:13]: sure

**Review by rishabh-arya95 — DISMISSED** [review]

File: pkg/urlmanager/urlmanager.go [hunk 3]
```diff
@@ -9,27 +9,17 @@ import (
 	"github.com/LambdaTest/synapse/pkg/global"
 )
 
-// GetDownloadURL returns file download url for given git provider
-func GetDownloadURL(gitprovider, repoSlug, commitID, fileName string) (string, error) {
```
Comment (palashmarantas) [inline pkg/urlmanager/urlmanager.go:13]: why this is removed ? How are we download raw files ?

File: pkg/global/nucleusconstants.go [hunk 4]
```diff
@@ -27,15 +27,11 @@ var FrameworkRunnerMap = map[string]string{
 	"jest":    "./node_modules/.bin/jest-runner",
 }
 
-// RawContentURLMap is map of git provider with there raw content url
-var RawContentURLMap = map[string]string{
```
Comment (palashmarantas) [inline pkg/global/nucleusconstants.go:31]: why this is removed . We used this to download .tas file .

File: pkg/urlmanager/urlmanager.go (hunk 3 above)
Comment (utkarsh-lambdatest) [reply pkg/urlmanager/urlmanager.go:13]: We are not using it anywhere

File: pkg/global/nucleusconstants.go (hunk 4 above)
Comment (utkarsh-lambdatest) [reply pkg/global/nucleusconstants.go:31]: It was also not being used anywhere

File: pkg/urlmanager/urlmanager.go (hunk 3 above)
Comment (utkarsh-lambdatest) [reply pkg/urlmanager/urlmanager.go:13]: https://github.com/LambdaTest/test-at-scale/pull/47#discussion_r822497660

File: pkg/urlmanager/urlmanager.go (hunk 2 above)
Comment (utkarsh-lambdatest) [reply pkg/urlmanager/urlmanager.go]: Somebody has written a unit test for it and that has been merged in dev, should I revert my deletion or should I take a pull and delete that unit test as well @saurabh-prakash ?

**Review by rishabh-arya95 — APPROVED** [review]

**Review by saurabh-prakash — APPROVED** [review]

**Pull request merged**